test/__tests__/infer-parser.js

```diff
@@ -0,0 +1,156 @@
+import { runCli } from "../utils";
+
+describe("stdin no path and no parser", () => {
+  describe("logs error and exits with 2", () => {
+    runCli("infer-parser/", [], { input: "foo" }).test({
+      status: 1,
+      stdout: "",
+      write: [],
+    });
+  });
+
+  // TODO (43081j): in prettier, this tests that it exits with 0 for
+  // whatever reason. should we do the same?
+  describe("--check logs error but exits with 1", () => {
+    runCli("infer-parser/", ["--check"], {
+      input: "foo",
+    }).test({
+      status: 1,
+      stdout: "",
+      write: [],
+    });
+  });
+
+  // TODO (43081j): in prettier, this tests that it exits with 0 for
+  // whatever reason. should we do the same?
+  describe("--list-different logs error but exits with 1", () => {
+    runCli("infer-parser/", ["--list-different"], {
+      input: "foo",
+    }).test({
+      status: 1,
+      stdout: "",
+      write: [],
+    });
+  });
+});
+
+describe("stdin with unknown path and no parser", () => {
+  describe("logs error and exits with 2", () => {
+    runCli("infer-parser/", ["--stdin-filepath", "foo"], {
+      input: "foo",
+    }).test({
+      status: 1,
+      stdout: "",
+      write: [],
+    });
+  });
+
+  // TODO (43081j): in prettier, this tests that it exits with 0 for
+  // whatever reason. should we do the same?
+  describe("--check logs error but exits with 1", () => {
+    runCli("infer-parser/", ["--check", "--stdin-filepath", "foo"], {
+      input: "foo",
+    }).test({
+      status: 1,
+      stdout: "",
+      write: [],
+    });
+  });
+
+  // TODO (43081j): in prettier, this tests that it exits with 0 for
+  // whatever reason. should we do the same?
+  describe("--list-different logs error but exits with 1", () => {
+    runCli(
+      "infer-parser/",
+      ["--list-different", "--stdin-filepath", "foo"],
+      { input: "foo" },
+    ).test({
+      status: 1,
+      stdout: "",
+      write: [],
+    });
+  });
+});
+
+describe("unknown path and no parser", () => {
+  describe("specific file is ignored", () => {
+    runCli("infer-parser/", ["--end-of-line", "lf", "FOO"]).test({
+      status: 0,
+      stdout: "",
+      write: [],
+    });
+  });
+
+  // TODO (43081j): this currently eats parser errors up, we need to decide
+  // if when !check && !list && !write we should still log parser errors to
+  // stderr
```

Comment:
we need to decide what to do about this too

if you run prettier without flags, that means `{check: false, list: false, write: false}`. that means this block of code does nothing:

https://github.com/prettier/prettier-cli/blob/d4d7ecaf76ea5888989b1aba2477c55960b871ba/src/index.ts#L181-L185

so we dont get output, but used to in old prettier

Reply:
Same as the other one I guess, let's consider ourselves as operating under a fourth mode and use that for branching.